src/jaegertracing/DynamicLoad.cpp

```diff
@@ -19,31 +19,37 @@
 
 #include <opentracing/dynamic_load.h>
 
-#include "TracerFactory.h"
+#include "jaegertracing/DynamicLoad.h"
 #include "jaegertracing/Tracer.h"
+#include "jaegertracing/TracerFactory.h"
 
-int OpenTracingMakeTracerFactory(const char* opentracingVersion,
-                                 const void** errorCategory,
-                                 void** tracerFactory)
+int JaegerTracingMakeTracerFactoryFct(const char* opentracing_version,
+                                           const char* opentracing_abi_version,
+                                           const void** error_category,
+                                           void* error_message,
+                                           void** tracer_factory)
 {
-    assert(errorCategory != nullptr);
-    assert(tracerFactory != nullptr);
+    assert(error_category != nullptr);
+    assert(tracer_factory != nullptr);
 #ifndef JAEGERTRACING_WITH_YAML_CPP
     *errorCategory =
         static_cast<const void*>(&opentracing::dynamic_load_error_category());
     return opentracing::dynamic_load_not_supported_error.value();
 #endif
-    if (std::strcmp(opentracingVersion, OPENTRACING_VERSION) != 0) {
-        *errorCategory = static_cast<const void*>(
+    if (std::strcmp(opentracing_version, OPENTRACING_VERSION) != 0 ||
+        std::strcmp(opentracing_abi_version, OPENTRACING_ABI_VERSION) != 0) {
+        *error_category = static_cast<const void*>(
```

Comment:
I don't think you want `std::strcmp(opentracing_version, OPENTRACING_VERSION) != 0` in the comparison.

The point of adding the ABI version was to allow the plugin to work with other versions of OT if there's no ABI breakage.

The version is only passed in for the case that you might have some sort of incompatible semantic change in the future.

Reply:
Good point.